jedeschule/spiders/saarland.py

```diff
@@ -11,45 +11,45 @@ class SaarlandSpider(SchoolSpider):
     ]
 
-    def parse(self, response):
-        tree = ET.fromstring(response.body)
+    def parse(self, response, **kwargs):
+        data = xmltodict.parse(response.text)
+        members = data.get("wfs:FeatureCollection", {}).get("wfs:member", [])
 
-        namespaces = {
-            "gml": "http://www.opengis.net/gml/3.2",
-            "SD": "https://geoportal.saarland.de/arcgis/services/Internet/Staatliche_Dienste/MapServer/WFSServer",
-        }
+        if not isinstance(members, list):
+            members = [members]
 
-        for school in tree.iter(
-            "{https://geoportal.saarland.de/arcgis/services/Internet/Staatliche_Dienste/MapServer/WFSServer}Schulen_SL"
-        ):
+        for member in members:
+            school = member.get("Staatliche_Dienste:Schulen_SL", {})
             data_elem = {}
-            for entry in school:
-                if (
-                    entry.tag
-                    == "{https://geoportal.saarland.de/arcgis/services/Internet/Staatliche_Dienste/MapServer/WFSServer}Shape"
-                ):
-                    # This nested entry contains the coordinates that we would like to expand
-                    lat, lon = entry.findtext(
-                        "gml:Point/gml:pos", namespaces=namespaces
-                    ).split(" ")
-                    data_elem["lat"] = lat
-                    data_elem["lon"] = lon
-                    continue
-                # strip the namespace before returning
-                data_elem[entry.tag.split("}", 1)[1]] = entry.text
+
+            for key, value in school.items():
+                if key == "Staatliche_Dienste:SHAPE":
+                    pos = (
+                        value.get("gml:Point", {})
+                        .get("gml:pos", "")
+                        .strip()
+                    )
+                    if pos:
+                        lat, lon = pos.split()
+                        data_elem["lat"] = float(lat)
+                        data_elem["lon"] = float(lon)
+                else:
+                    clean_key = key.split(":")[-1]
+                    data_elem[clean_key] = value
+
             yield data_elem
 
     @staticmethod
     def normalize(item: Item) -> School:
         # The data also contains a field called `SCHULKENNZ` which implies that it might be an id
         # that could be used, but some schools share ids (especially `0` or `000000`) which makes for collisions
-        id = item.get("OBJECTID")
+        school_id = item.get("OBJECTID")
 
         return School(
-            name=item.get("SCHULNAME"),
-            address=" ".join([item.get(part) for part in ["HNR", "STR_NAME"]]),
-            city=item.get("ORT_NAME"),
+            name=item.get("Bezeichnun"),
+            address=item.get("Straße", "").strip(),
+            city=item.get("Ort"),
             zip=item.get("PLZ"),
```

Comment:
It seems like the configuration of their WFS server is now wrong and returns the PLZ as a float where actually it should be a string (not even an integer since PLZs are always 5 digits in Germany and might start with zero). Can you parse to an integer here and pad with zeros (padding is strictly not necessary since the 0-leading zips are all in Sachsen afaik), please? 🙏